sdk/core/azure-core-http-vertx/src/test/java/com/azure/core/http/vertx/implementation/VertxRequestWriteSubscriberTests.java

```diff
@@ -0,0 +1,251 @@
+
+        waitForCompletion(promise);
+        assertTrue(promise.future().failed(), "The request should have failed.");
+        assertSame(writeError, promise.future().cause(), "The write error should be the cause of the failure.");
+        assertNull(onErrorDropped.get(), "No reactive error should be dropped.");
+    }
+
+    /**
+     * Tests that if the promise being used to manage the request-response state succeeds externally while writing the
+     * request it is handled appropriately.
+     * <p>
+     * The exception(s) that occur during writing should be dropped as the promise doesn't have a failure cause.
+     */
+    @Test
+    public void promiseIsSucceededExternalWhileRequestIsBeingSent() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        RuntimeException writeError = new RuntimeException("Write error");
+        RuntimeException reactiveError = new RuntimeException("Reactive error");
+        AtomicReference<Throwable> onErrorDropped = new AtomicReference<>();
+        Consumer<Throwable> onErrorDroppedConsumer = onErrorDropped::set;
+        Promise<HttpResponse> promise = Promise.promise();
+        Flux<ByteBuffer> data = Flux.generate(sink -> {
+            sink.next(ByteBuffer.wrap(DATA));
+            sink.error(reactiveError);
+        });
+        HttpClientRequest httpClientRequest = new MockHttpClientRequest() {
+            @Override
+            public void write(Buffer data, Handler<AsyncResult<Void>> handler) {
+                SharedExecutorService.getInstance().schedule(() -> {
+                    Promise<Void> promise = Promise.promise();
+                    promise.future().onComplete(handler);
+                    promise.fail(writeError);
+                    latch.countDown();
+                }, 1000, TimeUnit.MILLISECONDS);
+            }
+        };
+
+        VertxRequestWriteSubscriber subscriber = new VertxRequestWriteSubscriber(httpClientRequest, promise, null,
+            Context.of("reactor.onErrorDropped.local", onErrorDroppedConsumer));
+
+        data.subscribe(subscriber);
+        promise.complete();
+
+        if (!latch.await(5, TimeUnit.SECONDS)) {
+            fail("Timed out waiting for the request to be written.");
+        }
+
+        assertTrue(promise.future().succeeded(), "The request should have succeeded.");
+        assertEquals(writeError, onErrorDropped.get(), "The write error should be dropped.");
+        Throwable[] writeSuppressed = writeError.getSuppressed();
+        assertEquals(1, writeSuppressed.length, "There should be one suppressed exception.");
+        assertSame(reactiveError, writeSuppressed[0], "The reactive error should be the suppressed exception.");
+    }
+
+    private static void waitForCompletion(Promise<?> promise) throws InterruptedException {
+        Thread.sleep(1000);
+        for (int i = 0; i < 4; i++) {
+            if (promise.future().isComplete()) {
+                break;
+            }
+
```

Comment:
Confirming, do we meant to put the thread.sleep inside loop?

Reply:
Haha, good catch!